Refactor: Account Statement Report

jniles opened the pull request:
This commit implements multi-currency support on the Account Statement
Report. It is derived from a template by @WayneNiles.

The logic is somewhat complex.  If a user produces a report in the
enterprise currency, then the values should convert from a foreign
currency into the enterprise currency.  This is done through _division_
with the formula `amount / rate`.  However, if the user produces a
report in a different currency, all values must be converted from the
enterprise currency to that foreign currency.  This happens by
_multiplication_ with the formula `amount * rate`.

The report has been clarified by:
 1. Combining the two footers into one and naming it `footer`.
 2. Renaming the `openingBalance` property to `header`, since that is
 what is it.
 3. Removing the multiple options on the report page to simply give
 "currency" instead of having different total currencies.
 4. Renaming `accountDetails` to just `account`.

Finally, a new HBS helper has been made for rendering debits/credits
nicely.  It will automatically convert negative values into red currency
values.

Closes #1919.  Closes #2557.

See the attached example report: 
[ExampleReport.pdf](https://github.com/IMA-WorldHealth/bhima-2.X/files/1752025/ExampleReport.pdf)

jniles committed:
refactor(Account Report): support multi currency

This commit implements multi-currency support on the Account Statement
Report. It is derived from a template by @WayneNiles.

The logic is somewhat complex.  If a user produces a report in the
enterprise currency, then the values should convert from a foreign
currency into the enterprise currency.  This is done through _division_
with the formula `amount / rate`.  However, if the user produces a
report in a different currency, all values must be converted from the
enterprise currency to that foreign currency.  This happens by
_multiplication_ with the formula `amount * rate`.

The report has been clarified by:
 1. Combining the two footers into one and naming it `footer`.
 2. Renaming the `openingBalance` property to `header`, since that is
 what is it.
 3. Removing the multiple options on the report page to simply give
 "currency" instead of having different total currencies.
 4. Renaming `accountDetails` to just `account`.

Finally, a new HBS helper has been made for rendering debits/credits
nicely.  It will automatically convert negative values into red currency
values.

Closes #1919.  Closes #2557.

jniles committed:
refactor(patient): use debcred helper for amounts

This commit refators the patient financial report to use the debcred
helper to render the currency amounts in the report.

Related to #1919.

jniles committed:
feat(Account Statement Report): debit/credit sums

This commit adds sums to the debit and credit columns, but only if the
currency_id is the same throughout.

mbayopanda commented on client/src/modules/reports/generate/account_report/account_report.html:
```diff
@@ -43,55 +43,12 @@ <h3 class="text-capitalize" translate>REPORT.REPORT_ACCOUNTS.TITLE</h3>
               validation-trigger="ConfigForm.$submitted">
             </bh-date-interval>
 
-            <bh-hidden-field
```
The form is more clear now 👍

mbayopanda commented on server/lib/template/helpers/finance.js:
```diff
@@ -45,6 +46,22 @@ function indentAccount(depth) {
   return number ? number * INDENTATION_STEP : 0;
 }
 
+function debcred(value = 0, currencyId) {
```
🕺 🥇 really useful and well done

mbayopanda reviewed (approved):
The code and behaviors looks good to me.

I just noticed that i have zero display as `Taux` for `1$=930FC` when i choose Fc as currency

![account report](https://user-images.githubusercontent.com/5445251/36789224-0acdc76e-1c91-11e8-8ca4-b8842fac3f3f.PNG)

jniles committed:
fix(exchange): show HR rate instead of inverted

This commit fixes the exchange rate to show a human readable rate
instead of the inverted rate.  It now works correctly with both FC and
USD enterprise currencies.

mbayopanda commented on server/controllers/finance/reports/reportAccounts/index.js:
```diff
@@ -248,7 +248,7 @@ function getAccountTransactions(options, openingBalance = 0) {
         exchangedCumSum : lastCumSum,
         exchangedDate : new Date(),
-        invertedRate : util.roundDecimal(1 / totals.rate, 2),
+        invertedRate : Exchange.formatExchangeRateForDisplay(totals.rate),
```
👍
